### PR TITLE
add all pools, Batchnorm and Tanh (i.e. all ideeped MKLDNN ops) to MKLDNNFuser

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #{issue number}

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
Can you add all of the remaining mklddnn native ops in this PR?

**Comment by Krovatkin** [comment]
sure, I wanted to split them into two: pools and batchnorm + tanh for better readability, but one PR also works.

**Comment by eellison** [comment]
Yes, let’s just do it one go please. This would be the 4th or 5th pr adding native supported ops let’s get it done with.

**Comment by Krovatkin** [comment]
@eellison done!

File: test/jit/test_freezing.py [hunk 1]
```diff
@@ -1850,16 +1864,50 @@ def forward(self, x):
...
 
-            sub_model = torch.nn.Sequential(torch.nn.Conv2d(3, 64, 2, 2), torch.nn.AdaptiveAvgPool2d(4), torch.nn.Hardswish())
-            sub_model.eval()
-            mod = torch.jit.freeze(torch.jit.script(sub_model))
-            N, C, H, W, = 10, 3, 224, 224
-            inp = torch.randn(N, C, H, W)
-            self.run_pass("convert_frozen_ops_to_mkldnn", mod.graph)
-            self.assertTrue(torch.allclose(sub_model(inp), mod(inp)))
+            pooling_layers = [torch.nn.AdaptiveAvgPool2d(4),
+                              # torch.nn.AdaptiveMaxPool2d(4),
```
Comment (eellison) [inline test/jit/test_freezing.py:1871]: meant to be commented out?

File: test/jit/test_freezing.py [hunk 2]
```diff
@@ -37,6 +37,20 @@
     torch.ones(1).cuda()  # initialize cuda context
     TEST_CUDNN = TEST_CUDA and torch.backends.cudnn.is_acceptable(torch.tensor(1., device=torch.device('cuda:0')))
 
+def removeExceptions(graph):
+    def collectExceptions(exceptions, graph):
```
Comment (eellison) [inline test/jit/test_freezing.py:41]: this is just :
```
for exception in graph.findAllNodes("RaiseException"):
     exception.destroy()
```

File: test/jit/test_freezing.py [hunk 3]
```diff
@@ -1850,16 +1864,50 @@ def forward(self, x):
...
+                removeExceptions(mod.graph)
+                self.run_pass('dce', mod.graph)
+                self.run_pass("convert_frozen_ops_to_mkldnn", mod.graph)
+                self.assertTrue(torch.allclose(sub_model(inp), mod(inp)))
+
+    @unittest.skipIf(not torch._C.has_mkldnn, "MKL-DNN build is disabled")
+    def test_pool3d(self):
+        with set_default_dtype(torch.float):
+
+            pooling_layers = [  # torch.nn.AdaptiveAvgPool3d(4), NOT SUPPORTED
```
Comment (eellison) [inline test/jit/test_freezing.py:1893]: | torch.nn.AdaptiveAvgPool3d(4), NOT SUPPORTED

?

File: torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp [hunk 4]
```diff
@@ -811,12 +811,18 @@ class MKLDNNSubgraphSlicer {
       case aten::sigmoid:
       case aten::hardsigmoid:
       case aten::hardswish:
+      case aten::tanh:
+      case aten::batch_norm:
       // TODO: max_pool on mkldnn can be slower than in eager. ideally, we'd
       // only fuse it if we knew including max_pool lead to fewer layout
       // conversions. from initial testing including it speeds up models
       case aten::max_pool2d:
+      case aten::adaptive_max_pool2d: // not supported for floats
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp:820]: > not supported for floats 

If it's not supported for floats, why are we fusing it?

**Review by eellison — COMMENTED** [review]
one small question, looks good

File: torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp (hunk 4 above)
Comment (Krovatkin) [reply torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp:820]: oh snap, the `//` should've been on the other side!

File: test/jit/test_freezing.py (hunk 1 above)
Comment (Krovatkin) [reply test/jit/test_freezing.py:1871]: yes, ideep throws `operation isn't supported`

**Review by eellison — APPROVED** [review]
LGTM
